error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ComparatorRulesTestInput.java

```diff
@@ -77,32 +89,66 @@ ImmutableSet<Comparator<String>> testThenComparingNaturalOrder() {
         Comparator.<String>naturalOrder().thenComparing(s -> s));
   }
 
+  ImmutableSet<Integer> testCompareTo() {
+    return ImmutableSet.of(
+        Comparator.<String>naturalOrder().compare("foo", "bar"),
+        Comparator.<String>reverseOrder().compare("baz", "qux"));
+  }
+
+  int testMinOfVarargs() {
+    return Stream.of(1, 2).min(naturalOrder()).orElseThrow();
+  }
+
   ImmutableSet<String> testMinOfPairNaturalOrder() {
     return ImmutableSet.of(
+        "a".compareTo("b") <= 0 ? "a" : "b",
+        "a".compareTo("b") > 0 ? "b" : "a",
+        "a".compareTo("b") < 0 ? "a" : "b",
+        "a".compareTo("b") >= 0 ? "b" : "a",
+        Comparators.min("a", "b", naturalOrder()),
+        Comparators.max("a", "b", reverseOrder()),
         Collections.min(Arrays.asList("a", "b")),
         Collections.min(ImmutableList.of("a", "b")),
         Collections.min(ImmutableSet.of("a", "b")));
   }
 
   ImmutableSet<Object> testMinOfPairCustomOrder() {
     return ImmutableSet.of(
-        Collections.min(Arrays.asList(new Object(), new Object()), (a, b) -> -1),
-        Collections.min(ImmutableList.of(new Object(), new Object()), (a, b) -> 0),
-        Collections.min(ImmutableSet.of(new Object(), new Object()), (a, b) -> 1));
+        Comparator.comparingInt(String::length).compare("a", "b") <= 0 ? "a" : "b",
+        Comparator.comparingInt(String::length).compare("a", "b") > 0 ? "b" : "a",
+        Comparator.comparingInt(String::length).compare("a", "b") < 0 ? "a" : "b",
+        Comparator.comparingInt(String::length).compare("a", "b") >= 0 ? "b" : "a",
+        Collections.min(Arrays.asList("a", "b"), (a, b) -> -1),
+        Collections.min(ImmutableList.of("a", "b"), (a, b) -> 0),
+        Collections.min(ImmutableSet.of("a", "b"), (a, b) -> 1));
+  }
+
+  int testMaxOfVarargs() {
+    return Stream.of(1, 2).max(naturalOrder()).orElseThrow();
   }
 
   ImmutableSet<String> testMaxOfPairNaturalOrder() {
     return ImmutableSet.of(
+        "a".compareTo("b") >= 0 ? "a" : "b",
+        "a".compareTo("b") < 0 ? "b" : "a",
+        "a".compareTo("b") > 0 ? "a" : "b",
+        "a".compareTo("b") <= 0 ? "b" : "a",
+        Comparators.max("a", "b", naturalOrder()),
+        Comparators.min("a", "b", reverseOrder()),
         Collections.max(Arrays.asList("a", "b")),
         Collections.max(ImmutableList.of("a", "b")),
         Collections.max(ImmutableSet.of("a", "b")));
   }
 
   ImmutableSet<Object> testMaxOfPairCustomOrder() {
     return ImmutableSet.of(
-        Collections.max(Arrays.asList(new Object(), new Object()), (a, b) -> -1),
-        Collections.max(ImmutableList.of(new Object(), new Object()), (a, b) -> 0),
-        Collections.max(ImmutableSet.of(new Object(), new Object()), (a, b) -> 1));
+        Comparator.comparingInt(String::length).compare("a", "b") >= 0 ? "a" : "b",
```

Comment:
Hmm, usually we statically import calls to `comparingInt` because it is a candidate in `StaticImport`, but I think it's fine to leave it as is.

Reply:
Yeah, perhaps something we should resolve more generally. For this PR I tried to keep the changes minimal. (I did replace the `new Object()` calls, because I realized that with those we don't assert the proper order in the output template.)